### PR TITLE
Add e2e tests for triggers and dashboard

### PR DESCRIPTION
**Pull request by nikhil-thomas**
Fix `CheckDeployment` stage in `pkg/reconciler/common`: all reconcilers
were flaky as events were no requeued on deployment unavailable.

Add create, restore-deployments and delete e2e tests for TektonTriggers and TektonDashboard.

comment `aggregationRule` resource: ClusterRole, name: tekton-dashboard-extensions` to dashboard e2e tests and reconciler run without providing cluster admin privileges to operator sa. (https://github.com/tektoncd/operator/issues/131)

Comment 

Refactor TektonPipelines e2e tests

Signed-off-by: Nikhil Thomas <nikthoma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: test/presubmit-tests.sh.orig [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+source $(git rev-parse --show-toplevel)/vendor/github.com/tektoncd/plumbing/scripts/presubmit-tests.sh
+
+function check_go_lint() {
+    header "Testing if golint has been done"
+
+    # deadline of 5m, and show all the issues
+    golangci-lint -j 1 --color=never run
+
+>>>>>>> Stashed changes
```
Comment (vdemeester) [inline test/presubmit-tests.sh.orig:43]: There is a conflict here :stuck_out_tongue:

Comment (nikhil-thomas) [reply test/presubmit-tests.sh.orig]: :stuck_out_tongue: 
fixed :+1:

**Review by vdemeester — APPROVED** [review]

**Pull request merged**